semgrep/semgrep/synthesize_patterns.py

```diff
@@ -0,0 +1,26 @@
+import subprocess
+import tempfile
+from typing import List
+from typing import Optional
+
+import semgrep.config_resolver
+from semgrep.constants import PLEASE_FILE_ISSUE_TEXT
+from semgrep.constants import SEMGREP_PATH
+from semgrep.error import SemgrepError
+
+
+def synthesize_patterns(
+    language: str, code_to_synthesize: str, targets_str: List[str]
+) -> None:
+    targets = semgrep.config_resolver.resolve_targets(targets_str)
```

Comment:
We should probably raise an exception if targets is more than one file.

See dump_ast.py